.github/workflows/frontend.yml

```diff
@@ -1,15 +1,27 @@
-name: js build and lint
+name: Frontend
 
 on:
-  # Only runs webpack on master to save bundle size info into artifacts
-  # which are then used on PRs to compare against
   push:
     branches:
       - master
   pull_request:
 
 jobs:
+  files-changed:
+    name: detect what files changed
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check for frontend file changes
+        uses: getsentry/paths-filter@v2
+        id: changes
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-filters.yml
+
   typescript-and-lint:
+    needs: files-changed
```

Comment:
Once these FE jobs are not required we will be able to add `if: ${{ needs.changes.outputs.frontend == 'true' }}` and remove the in-steps conditionals.